Sources/Bag.swift

```diff
@@ -14,12 +14,22 @@ public struct Bag<Element> {
 		fileprivate let value: UInt64
 	}
 
-	fileprivate var elements: ContiguousArray<Element> = []
-	fileprivate var tokens: ContiguousArray<UInt64> = []
+	fileprivate var elements: ContiguousArray<Element>
+	fileprivate var tokens: ContiguousArray<UInt64>
 
-	private var nextToken = Token(value: 0)
+	private var nextToken: Token
 
-	public init() {}
+	public init() {
+		elements = ContiguousArray()
+		tokens = ContiguousArray()
+		nextToken = Token(value: 0)
+	}
+
+	public init<S: Sequence>(_ elements: S) where S.Iterator.Element == Element {
+		self.nextToken = Token(value: UInt64(elements.underestimatedCount))
```

Comment:
`underestimatedCount` is not guaranteed to equal the exact number of elements. We should derive this from the `ContiguousArray` constructed from `elements`.

Reply:
Ah, true.

Reply:
Amended:+1: